### ISSUE TITLE
DataRenderer.perform_is_valid_for_data can't be properly applied when dealing with typedefs

### ISSUE THREAD

**Issue by mostobriv** [Vector35/binaryninja-api#5485]
**Version and Platform (required):**
- Binary Ninja Version: 4.1.5396-dev Personal (716d21f3)
- OS: MacOS
- OS Version: 14.1

**Bug Description:**
Currently i can't tweak data rendering for types which is aliased (typedefd) to integer types (for example `typedef uint32_t rva_t;`) as in  `DataRenderer.perform_is_valid_for_data` callback for such types i got only `Type.IntegerType` instead of `Type.NamedTypeReference`, thus this type can't be distinguished from regular int types. Also `registered_name` for such type would be `None` so i can't use approach demonstrated in [this example](https://binary.ninja/2024/04/08/customizing-data-display.html)

**Steps To Reproduce:**
1. Create any typedef (for example `typedef uint32_t rva32_t;`)
2. Try to customise data rendering for it. Here is my testing snippet:
```python
class TestDataRender(DataRenderer):
    def __init__(self):
        super(TestDataRender, self).__init__()

    def perform_is_valid_for_data(self, ctx, view, addr, _type, context):
        # Equivalent of checking the platform
        if not view.platform:
            return False

        # print(context)
        print(_type.type_class)
        return False

    def perform_get_lines_for_data(self, ctx, view, addr, type, prefix, width, context):
        return []


TestDataRender().register_type_specific()
```

**Expected Behavior:**
I expect that typedefd type should be NamedTypeReference or at least should have its registered_name or any other way to be able to distinguish it from regular types (not structures).

**Additional Information:**
Actually i think that if it's not duplicate of #4552, then at least related to it, however Peter (plafosse) said that i should open separate issue, so here we go.

**Comment by xusheng6** [comment]
yeah this is probably not related to $4552, but they both expose issues in our handling for typedef-ed types

**Comment by ekilmer** [comment]
I'm also running into this issue. I'd like to render `time_t` values as human-readable dates.
